nemo/collections/asr/parts/k2/graph_decoders.py

```diff
@@ -0,0 +1,252 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+from omegaconf import DictConfig
+
+from nemo.collections.asr.parts.k2.classes import GraphIntersectDenseConfig
+from nemo.collections.asr.parts.k2.graph_compilers import CtcNumGraphCompiler, CtcTopologyCompiler
+from nemo.collections.asr.parts.k2.utils import (
+    create_supervision,
+    get_arc_weights,
+    invert_permutation,
+    load_graph,
+    make_blank_first,
+    prep_padded_densefsavec,
+    shift_labels_inpl,
+)
+from nemo.utils import logging
+
+# use k2 import guard
+# fmt: off
+from nemo.core.utils.k2_utils import k2_import_guard # isort:skip
+k2_import_guard()
```

Comment:
Should be an init time guard rather than import time

Reply:
Same, left it unchanged for now.